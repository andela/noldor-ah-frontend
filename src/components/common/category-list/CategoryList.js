import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { loadCategoriesList } from '../../../redux/actions/categoriesList/categoriesHeader';


export class CategoriesList extends Component {
  componentDidMount =() => {
    this.props.categoryList();
  }

  render() {
    const {
      categories
    } = this.props;
    return (
      <div className="container is-mt1 is-pb1">
        <div className="categories has-text-centered is-pt1">
          <ul>
            { categories.slice(0, 6).map((category, key) => (<li key={key}
              className="categories-list is-capitalized">
              {category}</li>))}
            <li className="is-paddingless">
              <button className=" categories-button theme-background">
          MORE</button></li>
          </ul>
        </div>

      </div>
    );
  }
}
CategoriesList.propTypes = {
  categories: PropTypes.array,
  categoryList: PropTypes.func
};

export const mapStateToProps = (state) => {
  return {
    categories: state.categoriesListReducer.categories
  };
};
export const mapDispatchToProps = {
  categoryList: () => loadCategoriesList(),
};
export default connect(mapStateToProps, mapDispatchToProps)(CategoriesList);
