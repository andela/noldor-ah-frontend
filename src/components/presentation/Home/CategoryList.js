import React from 'react';

const CategoryList = () => {
  return (
    <div className="container is-mt1 is-pb1 is-mb2">
      <div className="categories is-pt1">
        <ul>
          <li >music</li>
          <li>music</li>
          <li>music</li>
          <li>music</li>
          <li >music</li>

        </ul>

      </div>

    </div>

  );
};
export default CategoryList;
