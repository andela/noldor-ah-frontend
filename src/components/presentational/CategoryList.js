import React from 'react';

const CategoryList = () => {
  return (
    <div className="container is-mt1 is-pb1 is-mb2">
      <div className="categories is-pt1">
        <ul>
          <li >life</li>
          <li>music</li>
          <li>travel</li>
          <li>science</li>
          <li >finance</li>

        </ul>

      </div>

    </div>

  );
};
export default CategoryList;
