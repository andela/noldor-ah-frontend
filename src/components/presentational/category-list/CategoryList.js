import React from 'react';

const CategoryList = () => {
  return (
    <div className="container is-mt1 is-pb1 is-mb2">
      <div className="categories is-pt1">
        <ul>
          <li>LIFE</li>
          <li>MUSIC</li>
          <li>TRAVEL</li>
          <li>SCIENCE</li>
          <li>FINANCE</li>
        </ul>
      </div>
    </div>
  );
};
export default CategoryList;
