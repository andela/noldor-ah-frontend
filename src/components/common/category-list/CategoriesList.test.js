import { shallow } from 'enzyme';
import React from 'react';
import { CategoriesList, mapDispatchToProps, mapStateToProps } from './CategoryList';

describe('categories list should display', () => {
  const getProps = () => (
    {
      categories: ['MUSIC'],
      categoryList: () => {}
    }
  );

  const mounted = () => {
    const categories = getProps();
    return shallow(<CategoriesList {...categories} />);
  };
  it('snap shot', () => {
    const wrapper = mounted();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the categories', () => {
    const wrapper = mounted();
    expect(wrapper.find('.categories-list').text()).toEqual('MUSIC');
    expect(wrapper.find('.categories-button').text()).toEqual('MORE');
  });

  it('define mapStateToProps', () => {
    const state = { categoriesListReducer: { categories: '' } };
    expect(mapStateToProps(state)).toBeDefined();
  });

  it('define mapDispatchToProps', () => {
    // expect(mapDispatchToProps).toBeDefined();

    mapDispatchToProps.categoryList();
    expect(mapDispatchToProps.categoryList()).toBeDefined();
  });
});
