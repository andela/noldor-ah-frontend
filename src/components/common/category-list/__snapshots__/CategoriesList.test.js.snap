// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`categories list should display snap shot 1`] = `
<div
  className="container is-mt1 is-pb1"
>
  <div
    className="categories has-text-centered is-pt1"
  >
    <ul>
      <li
        className="categories-list is-capitalized"
        key="0"
      >
        MUSIC
      </li>
      <li
        className="is-paddingless"
      >
        <button
          className=" categories-button theme-background"
        >
          MORE
        </button>
      </li>
    </ul>
  </div>
</div>
`;
